CIP-XXXX/standard.cddl

```diff
@@ -0,0 +1,84 @@
+; Oracle Datum
+
+; The oracle datum is composed of a generic, shared and extended data.
+;   - With key 0: The generic data will be a list of different feed values.
+;   - With key 1: The shared data could not be part of the datum, but if it's
+;     present then carries the information about the timestamp and expiry of
+;     all the feeds.
+;   - With key 2: The extended data could not be part of the datum, but if it's
+;     present then carries some external information about the provider of the
+;     feeds.
+oracle_datum = #6.121(oracle_datum_list)
+oracle_datum_list =
+  [ ?  shared_data	
+  , 1* generic_data
+  , ?  extended_data
+  ]
+
+; At the moment we have two types of generic data:
+;   - price_data, for the exchange rate between two assets.
+;   - kyc_data, for “Know Your Customer” related data
+generic_data = price_data / kyc_data
+
+; A price_data entry normally defines the “Currency Pair” being priced.
+;  A “Currency Pair” is defined as “the first listed currency or base currency is
+;  bought while the second listed currency in a currency pair or quote currency is
+;  sold.” Each currency is defined separately in the entry, and referred to as the
+;  “base” and “quote” currency.
+price_data = #6.123([price_map])
+price_map =
+  { ? 0 : price        ; how many quote currency is received per base currency spent
+  , ? 1 : posixtime    ; unix timestamp related to when the price data was created
+  , ? 2 : posixtime    ; unix timestamp related to when the price data is expired
+  , ? 3 : precision    ; how many decimals of precision price is specified at
+  , ? 4 : asset_id     ; id of base
+  , ? 5 : asset_id     ; id of quote
+  , ? 6 : asset_symbol ; symbol of base
+  , ? 7 : asset_symbol ; symbol of quote
+  , ? 8 : asset_name   ; name of base
+  , ? 9 : asset_name   ; name of quote
+  }
+
+precision    = uint
+price        = uint
+asset_id     = uint
+asset_symbol = tstr
+asset_name   = tstr
+
+; In order:	
+; - Timestamp of the creation of all the generic data.
+; - Expiry of the creation of that data.
+shared_data = #6.121(shared_map)
+shared_map =
+  { ? 0: price_data
+  , ? 1: kyc_data
+  }
+
+; The extended data carries provider-specific information, and may
+;  be individually customized by oracle providers to provide data
+;  they deem relevant. It's thus not entirely defined by spec,
+;  but indices below 100 are "reserved" by the spec.
+extended_data = #6.122(extended_map)
+extended_map = 
+  { ? 0: oracle_provider_id
+  , ? 1: data_source_count
+  , ? 2: data_signatories_count
```

Comment:
As per the specification, the part you are now criticizing is `individually customized by oracle providers to provide data they deem relevant`, so what is generally applicable to everyone doesn't matter here. This is the part where each data provider would be putting things specific to their implementation, that does not fit into a general, shared standard.

Reply:
Oh, I see. So that part of your CDDL is just an example of how an oracle _could_ specify the relevant properties of how its consensus was reached on the data point?

I think my point still stands that the particular example (data source count and data signatories count) is insufficient, but it's not a blocker for your CIP if your CIP isn't mandating anything specific for oracle providers to put in there.

Reply:
Yes, that's correct. Our expectation would be that each oracle provider would be able to (through some reasonable process) simply declare "this is our custom properties" and it would be basically merged outright.

It would probably be subject to a low bar such as "being used in practice somewhere" just to keep things relevant, but the idea is to give each provider free reign here.

The underlying idea is that this would allow oracle providers to do their own thing in areas where they disagree with the standard, while still being able to conform to the standard in the areas they do agree. Like a safety release valve for disagreements around the standard, so that implementers are minimally constrained while still allowing maximum format sharing where consensus exist.